snewpdag/plugins/Chi2Calculation.py

```diff
@@ -0,0 +1,158 @@
+"""
+Chi2Calculator: generates a skymap of SN direction probabilities
+
+Constructor Arguments:
+    detector_list: list of strings, ["first_detector", "second_detector", ...]
+                the list of detectors that we want to include in the calculations
+                options: "HK", "IC", "JUNO", "KM3", "SK"
+    detector_location: csv file name ('detector_location.csv')
+    DET0: detector with the lowest error (delta_ts are calculated w.r.t. this detector) (e.g. 'SK')
+    NSIDE: (int) healpy map parameter, it describes map resolution (32 is a reasonable number)
+
+Output:
+    adds hp map (np.array) in nested ordering as 'chi2' and number of DOF (int) as 'ndof' to data
+"""
+
+import csv
+import logging
+import numpy as np
+import healpy as hp
+from scipy.stats import chi2
+from datetime import datetime
+
+from snewpdag.dag import Node
+
+
+class Chi2Calculator(Node):
+    def __init__(self, detector_list, detector_location,
+                 det0, NSIDE, **kwargs):
+        self.detector_info = {}
+        with open(detector_location, 'r') as f:
+            detectors = csv.reader(f)
+            for detector in detectors:
+                name = detector[0]
+                if name not in detector_list:
+                    continue
+                lon = np.radians(float(detector[1]))
+                lat = np.radians(float(detector[2]))
+                height = float(detector[3])
+                sigma = float(detector[4])
+                bias = float(detector[5])
+                if name == det0:
+                    self.d0_info = [
+                        lon, lat, height, sigma, bias, name]
+                else:
+                    self.detector_info[name] = [
+                        lon, lat, height, sigma, bias]
+        self.NSIDE = NSIDE
+        self.NPIX = hp.nside2npix(NSIDE)
+        self.precision_matrix = self.generatePrecisionMatrix()
+        self.ndof = len(detector_list) - 3
```

Comment:
perhaps the warning about ndof < 1 should be here in the constructor, rather than in each alert?